### ISSUE TITLE
[Submission] {50} params - {lichengliu03}

### ISSUE THREAD

**Issue by lichengliu03** [anadim/AdderBoard#11]
### Author

  lichengliu03

  ### Unique Parameter Count

  50

  ### Accuracy

  100% on 10,010 test pairs (10 edge + 10,000 random)

  ### Method

  Hand-coded weights

  ### Architecture

  1L custom GPT, d=4, 2 heads, hd=2, ReLU MLP, sinusoidal PE (period=11)

  ### Key Tricks

  - Factorized embedding: digit_values[10,1] * embed_dir[4]
  - Rotation-parameterized Q: 2 rotation angles instead of full Q projection
  - Shared K projection across heads, tied for Q extraction (weight tying)
  - Tied embed_dir for embedding + V extraction (weight tying)
  - Scalar V with manual attention (eliminates head_dim padding)
  - Rank-1 per-head O-projection: learned o_dir[2,4]
  - Direction-shared hat-function MLP with absorbed coefficients
  - Tied accum_dir for MLP output + head input direction (weight tying)
  - Parabolic head: 2 scalars generate all logits via lin*d*z + quad*d²
  - Fixed sinusoidal PE (period=11, 0p)
  
  ### Link to Code

  https://github.com/lichengliu03/TinyAdder-50p

  ### Verification Output

  ```shell
  Model: TinyAdder-50p
  Author: lliu22
  Parameters (unique): 50
  Architecture: 1L custom GPT, d=4, 2h, hd=2, ReLU, sin PE(period=11)
  Tricks: Factorized embedding: digit_values[10,1] * embed_dir[4], Tied
  embed_dir for embedding + V extraction (4p), Rotation-parameterized Q
  with K_proj-tied extraction (2p), Learned per-head O-projection
  directions (8p, absorbs o_scale), Tied accum_dir for MLP output + head
  input (4p), Absorbed carry/wrap coefficients into dir/bias rescaling,
  Parabolic head: 2 scalars (lin=2, quad=-1), Fixed sinusoidal PE
  (period=11, 0p)

    Progress: 1000/10010 (1000/1000 correct) [5.0s]
    Progress: 2000/10010 (2000/2000 correct) [10.0s]
    Progress: 3000/10010 (3000/3000 correct) [15.1s]
    Progress: 4000/10010 (4000/4000 correct) [20.1s]
    Progress: 5000/10010 (5000/5000 correct) [25.1s]
    Progress: 6000/10010 (6000/6000 correct) [30.1s]
    Progress: 7000/10010 (7000/7000 correct) [35.1s]
    Progress: 8000/10010 (8000/8000 correct) [40.2s]
    Progress: 9000/10010 (9000/9000 correct) [45.2s]
    Progress: 10000/10010 (10000/10000 correct) [50.2s]

  Results: 10010/10010 correct (100.00%)
  Time: 50.2s (199 additions/sec)
  Status: QUALIFIED (threshold: 99%)

  Additional Notes

  Parameter breakdown:
  digit_values        [10,1] =  10
  embed_dir              [4] =   4
  K_proj               [2,4] =   8
  q_angles               [2] =   2
  o_dir                [2,4] =   8
  carry_dir              [4] =   4
  wrap_dir               [4] =   4
  carry_bias             [2] =   2
  wrap_bias              [2] =   2
  accum_dir              [4] =   4  (tied: MLP output + head input)
  head_lin               [1] =   1
  head_quad              [1] =   1
  TOTAL                      =  50
